File: system-test/src/testkit/dsl/core/usecase/usecases/base/BaseBookShopUseCase.ts
import type { BookShopDriver } from '../../../../../driver/port/book-shop-driver.js';
import { BaseUseCase } from '../../../shared/base-use-case.js';

export abstract class BaseBookShopUseCase<TResponse, TVerification> extends BaseUseCase<
  BookShopDriver,
  TResponse,
  TVerification
> {}

File: system-test/src/main/java/com/jcupac/bookshop/testkit/dsl/core/usecase/usecases/base/BaseBookShopUseCase.java
package com.jcupac.bookshop.testkit.dsl.core.usecase.usecases.base;

import com.jcupac.bookshop.testkit.driver.port.BookShopDriver;
import com.jcupac.bookshop.testkit.dsl.core.shared.BaseUseCase;
import com.jcupac.bookshop.testkit.dsl.core.shared.UseCaseContext;

public abstract class BaseBookShopUseCase<R, V> extends BaseUseCase<BookShopDriver, R, V> {
    protected BaseBookShopUseCase(BookShopDriver driver, UseCaseContext context) {
        super(driver, context);
    }
}



